test.ts

```diff
@@ -480,6 +480,52 @@ describe('Project', function () {
     );
   });
 
+  it('adjusts peerDependencies of linked dependencies', function () {
+    let baseProject = new Project('base');
+    let alpha = baseProject.addDependency('alpha', {
+      files: {
+        'index.js': `
+          exports.betaVersion = function() {
+            return require('beta/package.json').version;
+          }
+          exports.gammaLocation = function() {
+            return require.resolve('gamma/package.json');
+          }
+        `,
+        deeper: {
+          'index.js': `
+            exports.betaVersion = function() {
+              return 'inner' + require('beta/package.json').version;
+            }
+          `,
+        },
+      },
+    });
+    alpha.pkg.peerDependencies = { beta: '^1.0.0' };
+    alpha.addDependency('gamma');
+    baseProject.addDependency('beta', { version: '1.1.0' });
+    baseProject.writeSync();
+
+    // precondition: in the baseProject, alpha sees its beta peerDep as beta@1.1.0
+    expect(require(require.resolve('alpha', { paths: [baseProject.baseDir] })).betaVersion()).to.eql('1.1.0');
+
+    let project = new Project('my-app');
+    project.linkDependency('alpha', { baseDir: baseProject.baseDir });
+    project.addDependency('beta', { version: '1.2.0' });
+    project.writeSync();
+
+    // in our linked project, alpha sees its beta peerDep as beta@1.2.0
+    expect(require(require.resolve('alpha', { paths: [project.baseDir] })).betaVersion()).to.eql('1.2.0');
```

Comment:
^

Reply:
This is test-suite only, I don't see any concern with module caching here.